### PR TITLE
+5 passing node:zlib tests

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

+5 passing node:zlib tests

- This fixes a memory leak in node:zlib caused by keeping strong references to the callbacks and the `this` value when it wasn't necessary
- This fixes the callbacks being associated with the wrong class (!)
- This fixes a memory leak that could potentially occur where the stream was initialized but never freed (or never closed)

### How did you verify your code works?

Ran existing node tests and verified they pass

### CODE REVIEW

**Comment by robobun** [comment]
<div><sup>Updated 8:22 AM PT - Dec 22nd, 2024</sup></div>

@Jarred-Sumner, your commit a62df5cdf9342c508a7c347d5c0d16156a23d362 is building: [`#8420`](https://buildkite.com/bun/bun/builds/8420)

File: test/js/node/test/parallel/test-internal-module-require.js [hunk 1]
```diff
@@ -1,112 +1,114 @@
...
+// //    is on.
+// // 2. We do not accidentally leak any modules to the public global scope.
+// // 3. Deprecated modules are properly deprecated.
 
 const common = require('../common');
 
-if (!common.isMainThread) {
-  common.skip('Cannot test the existence of --expose-internals from worker');
-}
+common.skip("This test is not going to be implemented in Bun. We do not support --expose-internals.")
```
Comment (nektro) [inline test/js/node/test/parallel/test-internal-module-require.js:12]: `common.skip` exits the process so the commenting out below here is unnecessary btw

File: src/bun.js/node/node_zlib_binding.zig [hunk 2]
```diff
@@ -662,16 +671,14 @@ pub const NativeBrotli = JSC.Codegen.JSNativeBrotli.getConstructor;
 
 pub const SNativeBrotli = struct {
     pub usingnamespace bun.NewRefCounted(@This(), deinit);
-    pub usingnamespace JSC.Codegen.JSNativeZlib;
+    pub usingnamespace JSC.Codegen.JSNativeBrotli;
```
Comment (nektro) [inline src/bun.js/node/node_zlib_binding.zig:674]: ! nice find

File: src/bun.js/node/node_zlib_binding.zig [hunk 3]
```diff
@@ -759,9 +769,12 @@ pub const SNativeBrotli = struct {
...
 
     pub fn deinit(this: *@This()) void {
-        this.write_callback.deinit();
-        this.onerror_value.deinit();
+        this.this_value.deinit();
         this.poll_ref.deinit();
+        switch (this.stream.mode) {
+            .BROTLI_ENCODE, .BROTLI_DECODE => this.stream.close(),
+            else => {},
+        }
```
Comment (nektro) [inline src/bun.js/node/node_zlib_binding.zig:777]:
```suggestion
        this.stream.close();
```

**Review by nektro — APPROVED** [review]
nice

File: src/bun.js/node/node_zlib_binding.zig (hunk 3 above)
Comment (Jarred-Sumner) [reply src/bun.js/node/node_zlib_binding.zig:777]: There's an assertion in there related to .mode so I was doing this to be cautious

**Pull request merged**